dataplane/updater.h

```diff
@@ -160,161 +142,69 @@ class updater_lpm4_24bit_8bit
 
 	void report(nlohmann::json& report) const
 	{
-		report["pointer"] = to_hex(pointer);
+		report["pointer"] = to_hex(pointer_.get());
 		report["extended_chunks_count"] = stats.extended_chunks_count;
 		report["extended_chunks_size"] = stats.extended_chunks_size;
 	}
 
-protected:
-	std::string name;
-	dataplane::memory_manager* memory_manager;
-	tSocketId socket_id;
-
-	object_type::stats_t stats;
-
-public:
-	object_type* pointer;
-};
-
-//
-
-class updater_lpm6_8x16bit
-{
-public:
-	using object_type = lpm6_8x16bit_atomic;
-
-	updater_lpm6_8x16bit(const char* name,
-	                     dataplane::memory_manager* memory_manager,
-	                     const tSocketId socket_id) :
-	        name(name),
-	        memory_manager(memory_manager),
-	        socket_id(socket_id),
-	        pointer(nullptr)
+	ObjectType* pointer()
 	{
-		stats.extended_chunks_count = 0;
-		stats.extended_chunks_size = 0;
-		stats.max_used_chunk_id = 0;
-		stats.free_chunk_cache.flags = 0;
+		return pointer_.get();
 	}
 
-	eResult init()
-	{
-		return create(2 * object_type::extended_chunks_size_min);
-	}
+private:
+	std::string name;
+	dataplane::memory_manager* memory_manager;
+	tSocketId socket_id;
+	stats_t stats;
+	dataplane::memory_manager::unique_ptr<ObjectType> pointer_;
 
-	eResult create(const uint64_t extended_chunks_size)
+	eResult Resize(const std::size_t size)
 	{
-		auto* next_pointer = memory_manager->create<object_type>(name.data(),
-		                                                         socket_id,
-		                                                         object_type::calculate_sizeof(extended_chunks_size));
-		if (next_pointer == nullptr)
+		memory_manager::unique_ptr<ObjectType> next = memory_manager->create_unique<ObjectType>(name.c_str(),
+		                                                                                        socket_id,
+		                                                                                        ObjectType::calculate_sizeof(size));
+		if (!next)
 		{
 			return eResult::errorAllocatingMemory;
 		}
 
-		object_type::stats_t next_stats;
+		stats_t next_stats;
 		next_stats.extended_chunks_count = 0;
-		next_stats.extended_chunks_size = extended_chunks_size;
+		next_stats.extended_chunks_size = size;
 		next_stats.max_used_chunk_id = 0;
 		next_stats.free_chunk_cache.flags = 0;
 
-		if (pointer)
-		{
-			next_pointer->copy(next_stats, stats, *pointer);
-			memory_manager->destroy(pointer);
-		}
-
+		next->copy(next_stats, stats, *(pointer_.get()));
 		stats = next_stats;
-		pointer = next_pointer;
-
+		std::swap(pointer_, next);
 		return eResult::success;
 	}
```

Comment:
And here we can do it like:
```cpp
eResult create(size_t size) {
        auto new_pointer = memory_manager->create_unique<ObjectType>(name.c_str(), socket_id, ObjectType::calculate_sizeof(size));
        if (!new_pointer) {
            return eResult::errorAllocatingMemory;
        }

        stats_t new_stats;
        new_stats.extended_chunks_count = 0;
        new_stats.extended_chunks_size = size;
        new_stats.max_used_chunk_id = 0;
        new_stats.free_chunk_cache.flags = 0;

        if (pointer_) {
            new_pointer->copy(new_stats, stats, *pointer_);
        }

        stats = new_stats;
        pointer_ = std::move(new_pointer);
        return eResult::success;
    }
```

```cpp
    eResult Resize(size_t new_size) {
        return create(new_size);
    }
```

Reply:
I really dislike that `create(size)` has a hidden logic of copying data without any mention or hint at call site.
Addition of separate `Resize(size)` is made specifically because of this.